### ISSUE TITLE
Parse the l0 FED_ENABLE_MASK and mask based on it

### ISSUE THREAD

**Issue by jhakala** [HCALRunControl/levelOneHCALFM#72]
The level0 passes us a list of FEDs which encodes whether they are "masked" (read: ignored) or not -- this parameter is stored in the HCALFM as the FED_ENABLE_MASK (or at least, this is what SHOULD be passed to the lvl1.) We should:
1) Verify that this parameter specifies which FEDs will be "maksed" (i.e. ignored by DAQ)
if so:
2) Add crates/related applications/entire FMs to the MASKED_APPs list based on this so that we don't initialize/configure executives whose FEDs are ignored in global.

**Comment by jhakala** [comment]
Upon further investigation, this isn't possible since the FED_ENABLE_MASK comes on configuring, after the ship has sailed.

**Comment by DryRun** [comment]
For my benefit, could I ask why the reason for closing this issue doesn't apply?

**Comment by jhakala** [comment]
Seth has found a miraculous way to mask on configuring -- basically we can use a new masking method where the level1 can mark a guy for destroying on configure now. This will only work for a global-style configuration (unlike masking on Initialize), but it does exactly what is needed for global operations where we want to mask our FMs based on the FED_ENABLE_MASK parameter.

PS: Seth's miracle code is here:
https://github.com/sethcooper/levelOneHCALFM/tree/testFMmaskingAtConfigureTime

**Comment by sethcooper** [comment]
From John:

I always saw there being three steps:
[1] Get the parameters from the PI. now we know they're FedInput* and FMMFeds
[2] do a check that the whole partition is masked by checking the FED_ENABLE_MASK
[3] do the magic configureMasking

**Comment by sethcooper** [comment]
(1) could also be accomplished by keeping a listing somewhere (e.g., XML file) of the FED to partition mapping.

**Comment by sethcooper** [comment]
We decided to specify the FED list as a configuration property in the template DB: one list per Level2 FM.

**Comment by jhakala** [comment]
The FED list configuration property is a string in JSON format. Here are some instructions for manipulating JSON in java:
http://stackoverflow.com/questions/2255220/how-to-parse-a-json-and-turn-its-values-into-an-array

**Comment by sethcooper** [comment]
The config property should be able to be extracted with this function:
https://github.com/HCALRunControl/levelOneHCALFM/pull/174

**Comment by jhakala** [comment]
Correction: the FED list should be a FunctionManager property, specific to individual level2s.

**Comment by sethcooper** [comment]
Is this finished now? Can this be closed? If we're only waiting on the cleaning, I would suggest moving that to a separate issue and closing this one.

**Comment by DryRun** [comment]
I believe so - @kakwok do you agree?

**Comment by kakwok** [comment]
FM side is done.
